### PR TITLE
fix: When we cant find some contentMedia dont fail. Just carry on.

### PR DESCRIPTION
**Pull request by johanwestin**
This solves the problem I've been having with Wordpress just showing an "Internal Server Error" when I try to execute specific queries.

### CODE REVIEW

**Review by jeanfredrik — CHANGES_REQUESTED** [review]
The array should probably be filtered to, so there’s no empty elements

**Comment by johanwestin** [comment]
I don't think the `preg_match_all` function can return empty items. 

`preg_match_all("/wp-(?:image|caption)-(\d+)/", $content, $matches);`

We are running `array_map` on `matches[1]`. 

That array will never contain empty items. the problem here is that the migration process is not 100% reliable and sometimes there are missing information in the database.

**Comment by jeanfredrik** [comment]
If any `$post` is empty, the array will contain a `null` element. Those should be filtered out. If `$post` is never empty, then we don’t need to change the code in the first place, right?

**Comment by johanwestin** [comment]
Array filter implemented here https://github.com/whitespace-se/wordpress-plugin-wp-graphql-extras/pull/1/commits/3a3d23872af1a0a8ae51ed21bdc40fa926f2616f

File: whitespace-wp-graphql-extras.php [hunk 1]
```diff
@@ -142,9 +142,9 @@ function ($type_registry) {
         preg_match_all("/wp-(?:image|caption)-(\d+)/", $content, $matches);
         $posts = array_map(function ($id) {
           $post = get_post($id);
-          return new Post($post);
+          return !empty($post) ? new Post($post) : null;
         }, array_unique($matches[1]));
-        return $posts;
+        return array_filter($posts, fn($i) => !empty($i));
```
Comment (jeanfredrik) [inline whitespace-wp-graphql-extras.php]: You can skip the second argument. The effect will be the same.

**Review by jeanfredrik — CHANGES_REQUESTED** [review]

File: whitespace-wp-graphql-extras.php (hunk 1 above)
Comment (johanwestin) [reply whitespace-wp-graphql-extras.php]: Done. Also removed the $posts variable. 

It was only confusing.

**Review by jeanfredrik — APPROVED** [review]

**Comment by jeanfredrik** [comment]
Released in [v0.1.2](https://packagist.org/packages/whitespace-se/wordpress-plugin-wp-graphql-extras#v0.1.2)

**Pull request merged**